minix/commands/autopart/autopart.c

```diff
@@ -1531,8 +1531,13 @@ void m_read(int ev, int *biosdrive)
 	stat_start(0);
 	fflush(stdout);
 
-	if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
-		if (device >= 0) { close(device); device= -1; }
+	device = open(curdev->name, O_RDWR, 0666);
+	if (device < 0) {
+		return;
+	}
+	else if (device >= 0) {
+		close(device);
+		device = -1;
```

Comment:
Yeah sorry this pull request kinda exploded. I'll do my best to split up my pull requests better next time. At this point, I'm just going to remove mode as this is already a bit of a mess. Sorry about that. Thank you for taking the time to work on this!